### DISCUSSION TITLE
Attemping to use github changelog type results in 404 when comparing tags

### DISCUSSION THREAD

**Discussion by richid** [goreleaser/goreleaser#4580] [Q&A] [answered]
Hello,

I am trying to convert my changelog generation type from `git` to `github` and I'm running into an issue. The release process fails with the following error:

>  ⨯ release failed after 2s                  error=GET https://api.github.com/repos/[my-org]/[my-repo]/compare/1.6.5...1.6.6?per_page=100: 404 Not Found []

In the case above the to-be-released version is `1.6.6` and the prior version is `1.6.5`. I believe this is erroring out because the `1.6.6` tag doesn't exist on the remote in GitHub yet. It's only been created locally, and therefore GitHub can't compare the tags. I verified this behavior with other, public repositories. This works: https://api.github.com/repos/octocat/Hello-World/compare/master...master but using non-existent refs doesn't: https://api.github.com/repos/octocat/Hello-World/compare/foo...bar

In case it matters, my organization is using GitHub Enterprise Cloud and I've verified that the token I am using is part of that org, so it should have access. It is scoped with the following permissions:
* `repo`
* `workflow`
* `write:packages`
* `read:org`
* `read:user`
* `user:email`
* `write:discussion`

I feel like I'm missing something obvious here. What am I doing wrong?

Thank you!

<details>
<summary>Full .goreleaser.yaml</summary>

```
project_name: <my-proj>
builds:
  - id: "lambda"
    dir: app/cmd/lambda
    binary: main
    flags:
      - -tags=dev
    env:
      - CGO_ENABLED=0
    goos:
      - linux
    goarch:
      - amd64
    ldflags:
      - "-s -w -X main.Version={{ .Version }}"
dockers:
  - id: lambda-validate
    goos: linux
    goarch: amd64
    dockerfile: docker/Dockerfile
    build_flag_templates:
      - "--build-arg=APP_PKG=cmd/lambda"
      - "--platform=linux/amd64"
    image_templates:
      - "<redact>.dkr.ecr.us-east-1.amazonaws.com/{{ .ProjectName }}/deploy:{{ .Version }}-{{ .ShortCommit }}"
      - "<redact>.dkr.ecr.us-east-1.amazonaws.com/{{ .ProjectName }}/deploy:latest"
    extra_files:
      - config/
changelog:
  use: github
  sort: asc
  groups:
    - title: Features
      regexp: "^.*feat:+.*$"
      order: 0
    - title: Bug fixes
      regexp: "^.*(bug|fix):+.*$"
      order: 1
    - title: Misc
      regexp: "^.*chore:+.*$"
      order: 2
release:
  footer: |
    **Full Changelog**: https://github.com/<my-org>/<my-proj>/compare/{{ .PreviousTag }}...{{ .Tag }}

**Comment by caarlos0** [comment, marked as answer]
yes, if the tag doesn't exist in github yet, there's not much we can do... it will not work indeed.


Will add a note to the docs about it.

**Comment by richid** [comment]
@caarlos0 thank you for the response!

I guess I'm confused as to how people are using this `github` changelog feature then. I think my use-case above is pretty basic and in that situation the tag will never exist since it's not pushed until the publish step.

**Reply by caarlos0** [reply]
afaik most people push the tag before running goreleaser

**Reply by richid** [reply]
Well this is embarrassing, it's right there in the [quick start docs](https://goreleaser.com/quick-start/) 🤦. I was just spoiled because creating a GitHub Release will automatically create a tag if it does not exist.

Thank you for setting me straight!

**Reply by caarlos0** [reply]
no problem :)
